scrapers/bank_of_zambia.py

```diff
@@ -1,48 +1,98 @@
+import re
+
 import requests
 from playwright.sync_api import sync_playwright
 from utils import (
+    calculate_checksum,
+    connect_circulars_to_associated_files,
     download_directory,
     get_browser,
     get_file_path,
     get_md5hash,
     get_standardized_filename,
     parse_arguments,
+    populate_file_dictionary,
+    zip_pdfs,
 )
 
 BASE_URL = "https://www.boz.zm/"
 CIRCULAR_URL = f"{BASE_URL}circulars.htm"
 
 
-def bank_of_zambia_scraper():
+def get_list_of_file_names_and_extensions(links):
+    file_names = []
+    # validate that each href has a file extension, split href on the '.' just before this extension, and add the resulting list to 'file_names'
+    for link in links:
+        href = link.get_attribute("href")
+        if not href:
+            raise ValueError("Invalid file URL.")
+        file_and_ext_list = href.rsplit(".", 1)
+        if len(file_and_ext_list) == 1:
+            raise ValueError("Link does not contain a file extension.")
+        file_names.append(file_and_ext_list)
+
+    return file_names
+
+
+def bank_of_zambia_scraper(n=10):
     with sync_playwright() as pw:
         browser = get_browser(pw)
         page = browser.new_page()
         page.goto(CIRCULAR_URL)
 
         session = requests.Session()
 
-        pdf_link = page.locator("div.list-group > a").all()[0]
-        link = pdf_link.get_attribute("href")
+        links = page.locator("div.list-group > a").all()
+        count = min(n, len(links))
 
-        if not link or not link.endswith(".pdf"):
-            browser.close()
-            raise ValueError("Invalid file URL.")
+        file_names_and_exts = get_list_of_file_names_and_extensions(links[6:9])
```

Comment:
Make sure we fix this part to account for `count`

```suggestion
        file_names_and_exts = get_list_of_file_names_and_extensions(links[:count])
```